File: frontend/src/api/answers.ts
import api from "./client";

export interface AnswerEvaluationResponse {
  relevance: number;
  depth: number;
  clarity: number;
  confidence: number;
  overall_score: number;
  feedback?: string | null;
}

export async function evaluateAnswer(params: {
  question: string;
  answer: string;
  target_role?: string;
}): Promise<AnswerEvaluationResponse> {
  const res = await api.post<AnswerEvaluationResponse>("/answers/evaluate", params);
  return res.data;
}

File: backend/app/api/routes/answers.py
from __future__ import annotations

from fastapi import APIRouter

from app.core.answer_evaluation import evaluate_answer
from app.schemas.answer_evaluation import (
    AnswerEvaluationRequest,
    AnswerEvaluationResponse,
)


router = APIRouter(prefix="/api/answers", tags=["answers"])


@router.post("/evaluate", response_model=AnswerEvaluationResponse)
def evaluate(payload: AnswerEvaluationRequest) -> AnswerEvaluationResponse:
    """
    Evaluate an interview answer for relevance, depth, clarity, and confidence.

    Uses Gemini when configured; otherwise falls back to a heuristic scorer.
    """
    return evaluate_answer(
        question=payload.question,
        answer=payload.answer,
        target_role=payload.target_role,
    )

